web/src/test/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImplTest.java

```diff
@@ -365,12 +361,82 @@ void shouldFilterDoctorIdsExcludedInFilter() {
 
             assertEquals(sickLeaves, actualSickLeaveList);
         }
     }
 
+    @Nested
+    class FilterOnRekoStatus {
+        List<SjukfallEnhet> sickLeaves;
+        RekoStatusDTO rekoStatus1;
+        RekoStatusDTO rekoStatus2;
+        SjukfallEnhet expectedSickLeave;
+        SjukfallEnhet anotherSickLeave;
+        SjukfallEnhet sickLeaveWithNoRekoStatus;
+
+        @BeforeEach
+        void setup() {
+            rekoStatus1 = new RekoStatusDTO();
+            rekoStatus1.setStatus(new RekoStatusTypeDTO(RekoStatusType.REKO_3.toString(), "Reko"));
+            rekoStatus2 = new RekoStatusDTO();
+            rekoStatus2.setStatus(new RekoStatusTypeDTO(RekoStatusType.REKO_2.toString(), "Reko"));
+            expectedSickLeave = createSjukFallEnhet(DIAGNOSIS_CODE, 5, PATIENT_ID);
+            expectedSickLeave.setRekoStatus(rekoStatus1);
+            anotherSickLeave = createSjukFallEnhet(ANOTHER_DIAGNOSIS_CODE, 12, ANOTHER_PATIENT_ID);
+            anotherSickLeave.setRekoStatus(rekoStatus2);
+            sickLeaveWithNoRekoStatus = createSjukFallEnhet(DIAGNOSIS_CODE, 5, PATIENT_ID);
+            sickLeaves = List.of(expectedSickLeave, anotherSickLeave, sickLeaveWithNoRekoStatus);
+        }
+
+        @Test
+        void shouldNotFilterOnRekoStatusIfNull() {
+            final var actualSickLeaveList = filterSickLeaves.filter(sickLeaves, null, Collections.emptyList(),
+                    null, null, null, TO_END_DATE, Collections.emptyList(), null);
+
+            assertEquals(sickLeaves, actualSickLeaveList);
+        }
+
+        @Test
+        void shouldNotFilterOnRekoStatusIfEmpty() {
+            final var actualSickLeaveList = filterSickLeaves.filter(sickLeaves, null, Collections.emptyList(),
+                    null, null, null, TO_END_DATE, Collections.emptyList(), Collections.emptyList());
+
+            assertEquals(sickLeaves, actualSickLeaveList);
+        }
+
+        @Test
+        void shouldFilterRekoStatusExcludedInFilter() {
+            final var actualSickLeaveList = filterSickLeaves.filter(sickLeaves, null, Collections.emptyList(),
+                    null, null, null, TO_END_DATE, Collections.emptyList(),
+                    Collections.singletonList(RekoStatusType.REKO_1.toString()));
+
+            assertEquals(1, actualSickLeaveList.size());
+            assertEquals(expectedSickLeave, actualSickLeaveList.get(0));
+        }
+
+        @Test
+        void shouldNotFilterRekoStatusIncludedInFilter() {
+            final var actualSickLeaveList = filterSickLeaves.filter(sickLeaves, null, Collections.emptyList(),
+                    null, null, null, TO_END_DATE, Collections.emptyList(),
+                    List.of(RekoStatusType.REKO_1.toString(), RekoStatusType.REKO_2.toString(), RekoStatusType.REKO_3.toString()));
+
+            assertEquals(sickLeaves, actualSickLeaveList);
+            assertEquals(expectedSickLeave, actualSickLeaveList.get(0));
```

Comment:
Redundant asserts? assertEquals(sickLeaves, actualSickLeaveList); already checks the return data.